qiime/core/type/primitive.py

```diff
@@ -349,6 +349,30 @@ def encode(self, value):
 Float = _Float('Float')
 
 
+class _Bool(_Primitive):
+    _valid_predicates = {Choices, Arguments}
+
+    def _is_element(self, value):
+        return isinstance(value, bool)
+
+    def _is_subtype_(self, other):
+        if (isinstance(other, type(Int)) and
+                self.predicate <= other.predicate):
+            return True
+        return super()._is_subtype_(other)
+
+    def decode(self, string):
+        if string in ("False", "0"):
+            return False
+        else:
+            return True
```

Comment:
Thinking more on it, right now `encode` and `decode` are implicitly JSON types, so it might make more sense to just match that exclusively.